### ISSUE TITLE
Crashing when reading undefined variable inside a iframe script

### ISSUE THREAD

**Issue by jhm-ciberman** [nwjs/nw.js#4912]
(NWJS v0.15.0 sdk stable win x64)
**Example code:**
```
"use strict";
var iframe = document.createElement("iframe");
document.body.appendChild(iframe);
setTimeout(()=>{ 
    var script = document.createElement('script');
    script.type = 'text/javascript';
    script.src = "my-script.js";
    iframe.contentWindow.document.head.appendChild(script);
},2500);
```
**Inside my-script.js:**
```
"use strict";
foo = UndefinedVariable; //The variable is undefined, the scripts loads fine and executes, 
// but nw.js crashes when the execution reach this point. No console.error is printed. Only crashes. 
```

**debug.log**
```
[0530/032119:ERROR:process_info.cc(608)] range at 0x10c740fe00000000, size 0x2ba fully unreadable
[0530/032119:ERROR:process_info.cc(608)] range at 0x10c7411e00000000, size 0x2ba fully unreadable
[0530/032119:ERROR:process_info.cc(608)] range at 0x0, size 0x2ba fully unreadable
[0530/032119:WARNING:in_range_cast.h(38)] value 2998168657146 out of range
[0530/032119:WARNING:in_range_cast.h(38)] value 2998168653368 out of range
[0530/032119:WARNING:in_range_cast.h(38)] value 2998168657186 out of range
[0530/032119:WARNING:in_range_cast.h(38)] value 108084205055182070 out of range
[0530/032119:WARNING:in_range_cast.h(38)] value 2998169270288 out of range

```

Thanks.

**Comment by Batcherinho** [comment]
Same here, have not yet found a way to get around it.

**Comment by andoitz** [comment]
I have same problem

**Comment by egoisticum** [comment]
Hi Dears,

Our company have the same problem.

The thing is, we are really pushed by deadlines, so we really have to fix it as soon as possible.


Would really appreciate your help here.

Thank you in advance.

**Comment by kresic-ivan-nsoft** [comment]
I would also like to see this fixed.

**Comment by rogerwang** [comment]
will fix it soon. Thanks.

**Comment by rogerwang** [comment]
This is fixed in git and will be available in the next nightly build.

**Comment by egoisticum** [comment]
@rogerwang you probably saved our project.
Can you let us know when we can expect next update, since this is very important issue for us?

Thank you in advance.

**Comment by frankhale** [comment]
@egoisticum, Roger said it would be available in the next nightly build. You can try the latest nightly build by downloading it from here: http://dl.nwjs.io/live-build/

**Comment by rogerwang** [comment]
Yes and here it is : http://dl.nwjs.io/live-build/06-03-2016/nw15-024592f-5c2963f-0b926c0-42b9671/v0.15.2/

0.15.2 will be released in next week. You can use this build if you cannot wait. The quality of this build is same as releases according to our tests.

**Comment by jhm-ciberman** [comment]
@rogerwang Thanks! I will try it right now with the plugin manager of my software and tell you. 
Btw, I am curious, (I don't know in depth C++) what was causing the error?

**Comment by rogerwang** [comment]
@jhm-ciberman the exception hander in blink and Node.js are catching this, but the one in Node.js failed to locate the corresponding Node environment and then it segfaults.

**Comment by vsemozhetbyt** [comment]
When will the fix be added to nw16-beta? Today live build still crashes.

**Comment by egoisticum** [comment]
Hi there,
we tested this build, but still getting same issue. So, my question is the same as it is question of @vsemozhetbyt

**Comment by Batcherinho** [comment]
This still seems to be happening on 0.15.2, any plans on revisiting the issue?

**Comment by rohan-deshpande** [comment]
I've just updated to `0.15.2` from `0.10.5` a couple of days ago and it's great apart from this issue. I'm not even launching in an iframe, the app randomly just crashes when it gets to an error and I can't find out what the error was because of course the console crashes with it. It doesn't seem to happen every time. 

I'll post my crash dump as soon as I figure out how to decode these things, bit difficult as well since they aren't timestamped or anything so I don't know which dump file to check. Also any reason why the `setCrashDumpDir` method was removed from the `gui.App` object? Seems like a pretty useful feature to have.

**Comment by iteufel** [comment]
@rohan-deshpande you can try 

    process.on('uncaughtException', function uncaughtExceptionHandler(err) {
      console.error(err);
      process.stdout.write(err.message);
    });

**Comment by rohan-deshpande** [comment]
Yep, I'm already doing this. I'm talking about nwjs literally crashing like, the whole app just quits.

**Comment by Batcherinho** [comment]
Is this something tied to win64 build exclusively?

I'm going to have to be pushy and ask @rogerwang for any kind of status update on the matter?
Our project is in production so it's quite troublesome, any news would be welcome.

**Comment by RohitSharma27** [comment]
Is there any debug.log file getting created in the directory containing nw.exe? For me, it's segmentation fault most of the times. It's window.open()'s asynchronous callback nature, which is giving me a hard time. It could be a native node module too, if you're using any.

**Comment by jhm-ciberman** [comment]
I have tried the nw Win x64 version and it works fine. :)

**Comment by frankhale** [comment]
@rohan-deshpande @Batcherinho do you have a small reproducible example which demonstrates the crashes?

**Comment by rohan-deshpande** [comment]
@frankhale hmm no, not right now, I do believe it has something to do with referencing undefined variables possibly. 

I'm using the Mac build so this is not Windows specific.

**Comment by frankhale** [comment]
@rohan-deshpande just trying to anticipate what @rogerwang is likely going to ask for. Does the example code in the original issue (1st post) work on the Mac build?

**Comment by rohan-deshpande** [comment]
@frankhale yeah I know. I'll try to reproduce my situation in a small sample. Will test the above code now and come back with results.

**Comment by rohan-deshpande** [comment]
This is still happening for me, randomly crashes when trying to log a variable that is undefined. Sometimes it crashes, sometimes it doesn't.

**Comment by freddy-daniel** [comment]
This bug is fixed on version 0.15.2 but appears on 0.16.0.

**Comment by rogerwang** [comment]
This should be a duplicate of #5148 and was fixed in git.
